src/script/hooks/useRoveFocus.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {useCallback, useState} from 'react';
+import {isKey, isTabKey, KEY} from 'Util/KeyboardUtil';
+import type {KeyboardEvent as ReactKeyboardEvent} from 'react';
+
+function useRoveFocus(size: number) {
```

Comment:
I'm not sure I understand what is `roveFocus` can you provide a JS doc explanation of what this does, that would be nice :)

Reply:
roving tabindex is when you have custom keyboard navigation afaik, in that case, navigating the sidebar with arrowkeys.

Reply:
@atomrc this one explained well https://www.freecodecamp.org/news/html-roving-tabindex-attribute-explained-with-examples/